### ISSUE TITLE
v2 Proposed Enhancement: fooTargetsChanged callback

### ISSUE THREAD

**Issue by leastbad** [hotwired/stimulus#336]
The ValueChanged callback has been a joy to work with, and it would be amazing to apply a similar mechanism to targets as well.

Having a `fooTargetsChanged()` callback would be good. `fooTargetAdded(element)` and `fooTargetRemoved(element)` would be great. Both would be sublime.

As an example, here is a controller that flashes the last target in a collection when it is inserted with a [CableReady](https://cableready.stimulusreflex.com) [insert_adjacent_html](https://cableready.stimulusreflex.com/usage/dom-operations/element-mutations#insertadjacenthtml) operation:

```js
import { Controller } from 'stimulus'
import { gsap } from 'gsap'
export default class extends Controller {
  static targets = ['location']
  connect () {
    this.element.addEventListener('cable-ready:after-insert-adjacent-html', this.locationInserted)
  }
  disconnect () {
    this.element.removeEventListener('cable-ready:after-insert-adjacent-html', this.locationInserted)
  }
  locationInserted = () => {
    gsap.fromTo(this.locationTargets[this.locationTargets.length - 1], { backgroundColor: "#FF0" }, { backgroundColor: "transparent" })
  }
}
```

If the proposed callback was implemented, this controller could be dramatically simplified:

```js
import { Controller } from 'stimulus'
import { gsap } from 'gsap'
export default class extends Controller {
  static targets = ['location']
  locationTargetAdded (element) {
    gsap.fromTo(element, { backgroundColor: "#FF0" }, { backgroundColor: "transparent" })
  }
}
```

**Comment by leastbad** [comment]
Talk to us, @panoply! I welcome other good ideas. How would you tackle this?

**Comment by dhh** [comment]
Now implemented through #350.
